client/src/components/pages/journal.tsx

```diff
@@ -0,0 +1,161 @@
+    }
+
+    setLoading(true);
+
+    try {
+      const response = await axios.post('http://localhost:8080/journal', { mood, entry });
+      console.log('Entry saved', response.data);
+      setEntry('');
+      setMood('');
+      setJournalEntries([response.data, ...journalEntries]);  // Prepend new entry
+    } catch (error) {
+      console.error('Error saving entry', error);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  // Prepare data for the pie chart
+  const moodCounts = journalEntries.reduce((counts: any, entry) => {
+    counts[entry.mood] = (counts[entry.mood] || 0) + 1;
+    return counts;
+  }, {});
+
+  const moodData = {
+    labels: Object.keys(moodCounts),
+    datasets: [
+      {
+        label: 'Mood Distribution',
+        data: Object.values(moodCounts),
+        backgroundColor: ['#FF6384', '#36A2EB', '#FFCE56', '#4CAF50', '#FF9800', '#8E44AD', '#E74C3C', '#2ECC71'],
+        borderColor: ['#FF6384', '#36A2EB', '#FFCE56', '#4CAF50', '#FF9800', '#8E44AD', '#E74C3C', '#2ECC71'],
+        borderWidth: 1,
+      },
+    ],
+  };
+
+  return (
+    <div className="journal-container">
+      <div className="tabs">
+        <button className={activeTab === 'journal' ? 'active' : ''} onClick={() => setActiveTab('journal')}>Journal Entry</button>
+        <button className={activeTab === 'chart' ? 'active' : ''} onClick={() => setActiveTab('chart')}>Mood Chart</button>
+        <button className={activeTab === 'entries' ? 'active' : ''} onClick={() => setActiveTab('entries')}>Past Entries</button>
+      </div>
+
+      {activeTab === 'journal' && (
+        <div className="journal-entry">
+          {/* Question of the Day */}
+          <div className="question-container">
+            <h3>Question of the Day</h3>
+            <p className="question-of-the-day">{questionOfTheDay}</p>  {/* Updated class name here */}
+          </div>
+          <textarea
+            className="journal-entry-textarea"
+            value={entry}
+            onChange={(e) => setEntry(e.target.value)}
+            placeholder="Write your response here..."
+          />
+          <div className="mood-selection">
+            {moods.map((moodObj, index) => (
+              <button
+                key={index}
+                className={`mood-button ${mood === moodObj.emoji ? 'selected' : ''}`}
+                onClick={() => setMood(moodObj.emoji)}
+              >
+                {moodObj.emoji} {moodObj.name}
+              </button>
+            ))}
+          </div>
+
+          <button
+            className="save-entry-button"
+            onClick={handleSaveEntry}
+            disabled={!mood || !entry || loading}
+          >
+            {loading ? 'Saving...' : 'Save Entry'}
+          </button>
+        </div>
+      )}
+
+      {activeTab === 'chart' && (
+        <div className="mood-chart">
+          <h3>Mood Distribution</h3>
+          <Pie data={moodData} />
```

Comment:
the pie chart is comically large loll, and the legend is very tiny

![image](https://github.com/user-attachments/assets/3e31c16f-f632-46ab-854f-de735f3cd3bf)

Reply:
fixed lol

<img width="441" alt="Screenshot 2024-12-04 at 8 26 04 AM" src="https://github.com/user-attachments/assets/5595f529-e0ad-4a03-a42b-97a66781263f">